packages/uploader-core/src/index.ts

```diff
@@ -1,13 +1,17 @@
-import { Link, Version } from 'multiformats'
-import type { CARMetadata, Service } from '@web3-storage/upload-client/types'
+import type { CARMetadata, Service, ProgressFn, ProgressStatus, FetchOptions } from '@web3-storage/upload-client/types'
 import type { ConnectionView, Principal } from '@ucanto/interface'
 
+import { Link, Version } from 'multiformats'
+
 export { uploadFile, uploadDirectory } from '@web3-storage/upload-client'
 
-export type { CARMetadata, Service }
+export type { CARMetadata, Service, ProgressFn, ProgressStatus, FetchOptions }
+
+export type ProgressStatuses = Record<string, ProgressStatus>
```

Comment:
from the perspective of this library, I don't really think so - in practice it's usually going to be the URL of the upload whose progress is being reported, but all we care about is that it uniquely identifies a particular upload so that it can be distinguished from other uploads we are getting progress reports on...

Reply:
```suggestion
/**
  * Upload identifier that can be used to distinguish it from other uploads in progress.
  * In practice this will be an upload URL if it is unique per upload.
  */
export type UploadID = string
/**
 * Mapping of upload identifiers to corresponding upload status.
 */
export type ProgressStatuses = Record<UploadID, ProgressStatus>
```

Given the comments above I suggest above changes